File: api/src/main/java/io/bunnies/baas/resources/types/WebmMediaType.java
package io.bunnies.baas.resources.types;

import io.bunnies.baas.Constants;

public class WebmMediaType extends ResourceMediaType {
    public static final String KEY = "WEBM";

    public WebmMediaType(String resourceID) {
        super(resourceID);
    }

    @Override
    public String constructUrl() {
        return Constants.MEDIA_BASE_URL + "webm/" + this.resourceID + ".webm";
    }
}

File: api/src/main/java/io/bunnies/baas/resources/types/WebmMediaType.kt
package io.bunnies.baas.resources.types

class WebmMediaType(resourceID: String, mediaBaseUrl: String) : ResourceMediaType(resourceID, mediaBaseUrl) {

    override fun constructUrl(): String {
        return this.mediaBaseUrl + "webm/" + this.resourceID + ".webm"
    }

    companion object {
        val KEY = "WEBM"
    }
}
